applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/adjoint_structural_response_function.cpp

```diff
@@ -52,8 +49,13 @@ namespace Kratos
 
         if(mGradientMode == 1)
         {
-            VariableUtils().SetNonHistoricalVariable(PERTURBATION_SIZE, mDelta, mrModelPart.Elements());
-            VariableUtils().SetNonHistoricalVariable(PERTURBATION_SIZE, mDelta, mrModelPart.Conditions());
+            double delta = mResponseSettings["step_size"].GetDouble();
+            mrModelPart.GetProcessInfo()[PERTURBATION_SIZE] = delta;
```

Comment:
here the naming is inconsistent
delta & PERTURBATION_SIZE & step_size

Reply:
I changed the delta, which indeed does not make sense. About the difference PERTURBATION_SIZE and step_size, see @armingeiser comment above.